### PR TITLE
Stat tracker eliza final

### PR DESCRIPTION
**Pull request by elizakeating**
## Description

Added StatTracker functionality for the SeasonStatistics methods. Fixed SeasonStatistics tests and methods to pass spec harness. Fixed 2 failing tests for LeagueStatistics.

## Type of change

- [x] fix
- [x] feat
- [ ] test
- [x] refactor
- [ ] docs

## Checklist

- [x] code has been self reviewed
- [x] code runs without any errors
- [x] thorough testing has been implemented if adding feature
- [x] all tests pass

### Thanks!

### CODE REVIEW

**Review by ethanb1145 — COMMENTED** [review]
Thanks for adding more to the functionality of the StatTracker! Good to see that you were able to tweak your methods a bit so that they could pass both the individual tests and the spec harness. Good job !

File: lib/season_statistics.rb [hunk 1]
```diff
@@ -15,49 +15,50 @@ def winningest_coach(season_id)
...
+        winning_games[row[:head_coach]] += 1.0
+        games_played[row[:head_coach]] += 1.0
       elsif row[:result] == "LOSS"
-        winning_games[row[:team_id]] = 0
-        games_played[row[:team_id]] += 1
+        winning_games[row[:head_coach]] += 0.0
+        games_played[row[:head_coach]] += 1.0
+      elsif row[:result] == "TIE"
+        winning_games[row[:head_coach]] += 0.0
+        games_played[row[:head_coach]] += 1.0
```
Comment (kbug819) [inline lib/season_statistics.rb:30]: Great catch on these changes!

**Review by nurudeensulaimon — COMMENTED** [review]
Reviewed. Great job on refactoring. We should be able to have the few errors resolved. Thanks!

**Comment by kbug819** [comment]
Awesome job on this! Passing Stat Tracer!

**Pull request merged**